Full setup db

williamh890 opened the pull request.

williamh890 committed:
Merge branch 'dev' of https://github.com/asfadmin/hyp3-in-a-box into dev

williamh890 committed:
Remove extraneous text from email

williamh890 committed:
Admin user and default processes added in setupdb lambda

williamh890 committed:
Working on custom cloudformation resource to generate passwords for template

williamh890 committed:
Adding pipfile and requirements.txt to setup_db lambda

williamh890 committed:
Moving random_pass into lambdas folder

williamh890 committed:
Refactor random_password custom resource function into class

williamh890 committed:
Move custom_resource into shared module

Added
  - Test
  - Better runtime validation of base classes

williamh890 committed:
Random password lambda uses custom_resource module

williamh890 committed:
random_pass template uses python3.6+fix wrong GetAtt call

williamh890 committed:
Remove old autoclass in docs and small conf.py refactor

williamh890 committed:
Add requirements for random_password lambda

williamh890 committed:
Merge branch 'dev' of https://github.com/asfadmin/hyp3-in-a-box into full-setup-db

Had to fix setup_db testing as they referenced os.environ variables
these are now properly mocked to point to unittest db when testing

williamh890 committed:
Remove extra session.commit calls

Askaholic commented on cloudformation/tropo/templates/hyp3_random_pass.py:
```diff
@@ -0,0 +1,50 @@
+
```
Where are the docstrings!!??

Askaholic commented on cloudformation/tropo/templates/hyp3_rds.py:
```diff
@@ -44,7 +44,6 @@ def get_security_group(name: str):
 # https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.DBInstanceClass.html
 hyp3_db = t.add_resource(rds.DBInstance(
     "Hyp3DB",
-    DBInstanceIdentifier="hyp3-in-a-box",
```
Why do you hate names so much? If I am going to spin up someone else's cloudformation template in my AWS Account, I want to be able to clearly see where resources are coming from. I don't want to have 5 RDS's all named `uihnkbvgcyitsudfg`. If name conflicts are an issue, then just make it a parameter with default value.

williamh890 replied:
I just took that out cuz I was lazy and it was erroring the stack, I was planning on making an environment variable to use parameterized names or not in the template.

Askaholic commented on lambdas/random_password/src/random_password.py:
```diff
@@ -0,0 +1,46 @@
+        characters = string.digits + string.ascii_letters
+
+        random_string = ''.join(
+            random.SystemRandom().choice(characters) for _ in range(length)
```
WOw s0 SEcUr3!

williamh890 replied:
lel

Askaholic commented on lambdas/setup_db/src/init_db.py:
```diff
@@ -8,30 +8,76 @@
+    add_user_sql = text(f"""
+        CREATE USER hyp3_user WITH PASSWORD :password;
+        GRANT ALL PRIVILEGES ON DATABASE {db_name} to {user};
+    """.format(db, user))
```
You never want to do string formatting on something which will later be run as SQL. This is how SQL injections happen! Always use prepared statements, and parameters.

williamh890 replied:
It throw an error when using the prepared statement version. This was a hack to get the merge to pass the tests.

Askaholic commented on lambdas/setup_db/src/init_db.py:
```diff
@@ -8,30 +8,76 @@
     Base.metadata.create_all(db.engine)
+
+
+def make_hyp3_admin_user(db):
```
Presumably this will be parameterized later. Maybe add a TODO comment as a reminder.

williamh890 committed:
Wrapping setup_db lambda with custom resource class

This way it can be invoked during the creation of the cloudformation
template.

williamh890 committed:
Move db connection inside of CustomResource class

This way it won't connect to db during delete operations

williamh890 committed:
Move base lambda function template code into function

- made names into a environment parameter
- made database creation into a environment parameter

The only function using the utils base lambda template is setup_db.

williamh890 committed:
Property testing for create_stack.py

williamh890 committed:
Debug printouts in setup_db

williamh890 committed:
Format the created user in both places

williamh890 committed:
Use double quotes

williamh890 committed:
Test now fully exersices setup_db function

williamh890 committed:
Change import to import http.client explicitly

williamh890 committed:
Fix json.dump to json.dumps

williamh890 committed:
Remove extra .format in init_db

The .format is needed because when sqlalcemy text formatting the values
are wrapped in quotes, which throws a programming error.

williamh890 committed:
Use filename argument in cloudformation-config.sh

williamh890 committed:
Hyp3 admin user accounts now takes params+processes loaded from file

- json file in s3 with all info about hyp3 default processes
  setup_db uses this file to create processes in db.
- params for username and email for the admin hyp3 user

williamh890 committed:
Folder where processing glue code will live

Right now just contains notify only stuff

williamh890 committed:
Update docs for random pass

williamh890 committed:
Fix codefactor issues
